pandas/tests/indexing/test_categorical.py

```diff
@@ -564,89 +576,82 @@ def test_reindexing(self):
 
         # reindexing
         # convert to a regular index
```

Comment:
make a new test function(s), create the inputs in-line

Reply:
The test is quite big, but I didn't make it any bigger. I could split it up at the cost of duplicating a bit of code (mostly just the test data). Is that what you're after?

Reply:
not necessarily part of this PR, but if this test can be split up into individual tests with clearer targets that'd be  helpful